lib/action_subscriber/message_retry.rb

```diff
@@ -0,0 +1,51 @@
+module ActionSubscriber
+  module MessageRetry
+    def self.redeliver_message_with_backoff(env)
+      ttl = calculate_ttl(env)
+      return unless ttl
+      with_exchange(env, ttl) do |exchange|
+        exchange.publish(env.encoded_payload,
+          :content_type => env.content_type,
+          :routing_key => env.routing_key,
+          :expiration => ttl
+        )
+      end
+    end
+
+    # Private Implementation
+    def self.calculate_ttl(env)
+      last_ttl = get_last_ttl_or_default(env.headers["x-death"], 20)
+      next_ttl = last_ttl * 5
+      return false if next_ttl > 86_400_000
+      next_ttl
+    end
+
+    def self.get_last_ttl_or_default(x_death, default)
+      return default unless x_death
```

Comment:
I implemented the parsing logic here locally by only handling the simple cases (ie. this doesn't parse all the keys and values out of the header, but it covers the ones we care about)